test/framework/management/kind/mgmt.go

```diff
@@ -0,0 +1,239 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kind
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io/ioutil"
+	"os"
+
+	"github.com/pkg/errors"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	"sigs.k8s.io/cluster-api/test/framework/exec"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// Shells out to `kind`, `kubectl`
+
+// Cluster represents a Kubernetes cluster used as a management cluster backed by kind.
+type Cluster struct {
+	Name              string
+	KubeconfigPath    string
+	Client            client.Client
+	Scheme            *runtime.Scheme
+	WorkerKubeconfigs map[string]string
+	// TODO: Expose the RESTConfig and a way to create a RESTConfig for the workload clusters for static-client uses
+	//       (pod logs, exec and any other subresources)
+}
+
+// NewCluster sets up a new cluster to be used as the management cluster.
+func NewCluster(name string, scheme *runtime.Scheme) (*Cluster, error) {
+	createCmd := exec.NewCommand(
```

Comment:
Thanks for pushing here. Let me play around with the API and see if it exposes everything I need. getting rid of the exec call would be great

Reply:
Because the docker provider is using an older version of the kind library, this module will have to either use that older version (match module versions) or we will have to clean up capd to use the new kind library (preferred). 

Either way the APIs are *not* compatible and because they are the same module we cannot import both versions of the same module :(.

I don't think implementing this using the older kind api makes sense because the API is completely different now and we'd likely want to keep that dependency up to date.

Therefore, the steps to take after this merges would be:

1. reimplement capd using kind v0.5.2 API (this is actually a ton of work, it would probably take a couple of days to get it right)
2. Use the new kind library here in the kind mgmt cluster to remove the dependency on the kind binary.

So to fix this problem

Reply:
Please see https://github.com/kubernetes-sigs/cluster-api/issues/1751